libraries/chain/controller.cpp

```diff
@@ -2146,13 +2146,18 @@ struct controller_impl {
 
       if (conf.truncate_at_block > 0 && chain_head.is_valid()) {
          if (chain_head.block_num() == conf.truncate_at_block && fork_db_has_root()) {
-            fork_db_.apply<void>([&](auto& fork_db) {
-               if (auto head = fork_db.head(); head && head->block_num() > conf.truncate_at_block) {
-                  ilog("Removing blocks past truncate-at-block ${t} from fork database with head at ${h}",
-                        ("t", conf.truncate_at_block)("h", head->block_num()));
-                  fork_db.remove(conf.truncate_at_block + 1);
-               }
-            });
+            if (fork_db_.version_in_use() == fork_database::in_use_t::both) {
+               // in savanna transition
+               wlog("In the middle of Savanna transition, truncate-at-block not allowed, ignoring truncate-at-block ${b}", ("b", conf.truncate_at_block));
```

Comment:
`nodeos_read_terminate_at_block_test` was using the `truncate` to resolve https://github.com/AntelopeIO/spring/issues/1365. Should `nodeos_read_terminate_at_block_test` be updated to reflect the fact that forkdb is not removed during Savanna transition?

Reply:
Maybe, but my thought is it might be so rare of a failure that we actually never trip it. You need both to catch it in the middle of a Savanna transition and also when it has added an additional entry in the fork database so that HEAD moves on restart. Both are relatively rare.